torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -304,5 +333,39 @@ void ScalarTypeAnalysisNodeForONNX(Node* n) {
   ImplicitCastNodeForONNX(n);
 }
 
+static void ImplicitCastNodeForONNXWithoutLowPrecision(Node* n) {
+  if (IsImplicitCastSupported(n->kind())) {
+    auto expected_scalar_type = InferExpectedScalarType(n);
+    if (expected_scalar_type) {
+      UpdateScalarTypeForInputs(n, *expected_scalar_type);
```

Comment:
It's bad design duplicating code like this. What if next time some updates need to happen for `ImplicitCast`? Now there are two places to look for changes.

Reply:
Thanks, redesigned the code. Add the opset_version and standardOps_lowprecision_cast parameter.